src/uci.cpp

```diff
@@ -395,4 +407,12 @@ Move UCI::to_move(const Position& pos, string& str) {
   return MOVE_NONE;
 }
 
+void UCI::terminate_on_critical_error(const std::string& fullCommand, const std::string& message) {
+  // Ideally we would report the error and continue, but UCI does not define such a situation nor
+  // gives any guarantees as to the program's behaviour for the user to rely on,
+  // so the safest option is to terminate.
+  sync_cout << "info string CRITICAL ERROR: Command `" << fullCommand << "` failed. Reason: " << message << '\n' << sync_endl;
+  std::terminate();
```

Comment:
Exiting is an appropriate action here, but I don’t think `std::terminate()` is the right way to do it.  `std::terminate()` typically results in a core dump, so it should only be used when there is a bug in Stockfish itself.  It isn’t the appropriate reaction to invalid input.  In this case, I would just call `exit()` if there are no other threads running, or `quick_exit()` or `_Exit` otherwise.